backend/lib-shared-site-api/src/cache/cache.rs

```diff
@@ -263,4 +243,24 @@ impl AppCache {
     pub async fn remove_metadata(&self, site_id: &str) {
         self.metadata_cache.invalidate(site_id).await;
     }
+
+    // Site Data
+    pub async fn get_site_with(
```

Comment:
@ClaireMao1996 I ended up splitting the site data cache from usage. There were some bugs in publishing and creating drafts, and I think the code is a little easier to follow this way. Instead of updating the cache when the site data changes, we can just clear it and let `get_current_site` refresh it when needed.